stages/test/test_erofs.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/python3
+
+import os.path
+import subprocess
+from unittest import mock
+
+import pytest
+
+import osbuild.meta
+from osbuild.testutil.imports import import_module_from_path
+
+
+def make_fake_input_tree(tmpdir, fake_content: dict) -> str:
+    basedir = os.path.join(tmpdir, "tree")
+    for path, content in fake_content.items():
+        dirp, name = os.path.split(os.path.join(basedir, path.lstrip("/")))
+        os.makedirs(dirp, exist_ok=True)
+        with open(os.path.join(dirp, name), "w", encoding="utf-8") as fp:
+            fp.write(content)
+    return basedir
+
+
+def have_erofs() -> bool:
+    try:
+        r = subprocess.run(
+            ["mkfs.erofs", "--help"], check=False, stderr=subprocess.DEVNULL)
+    except FileNotFoundError:
+        return False
+    return r.returncode == 0
+
+
+@pytest.mark.skipif(not have_erofs(), reason="no mkfs.erofs")
+def test_erofs_integration(tmp_path):
+    fake_input_tree = make_fake_input_tree(tmp_path, {
+        "/file-in-root.txt": "other content",
+        "/subdir/file-in-subdir.txt": "subdir content",
+    })
+    inputs = {
+        "tree": {
+            "path": fake_input_tree,
+        }
+    }
+    stage_path = os.path.join(os.path.dirname(__file__), "../org.osbuild.erofs")
+    stage = import_module_from_path("erofs_stage", stage_path)
+    filename = "some-file.img"
+    options = {
+        "filename": filename,
+    }
+
+    stage.main(inputs, tmp_path, options)
+
+    img_path = os.path.join(tmp_path, "some-file.img")
+    assert os.path.exists(img_path)
+    output = subprocess.check_output([
+        "dump.erofs", "--ls", "--path=/", img_path], encoding="utf-8")
+    assert "subdir\n" in output
+    assert "file-in-root.txt\n" in output
+    output = subprocess.check_output([
+        "dump.erofs", "--ls", "--path=/subdir", img_path], encoding="utf-8")
+    assert "file-in-subdir.txt\n" in output
+
+
+@mock.patch("subprocess.run")
+@pytest.mark.parametrize("test_options,expected", [
+    ({}, []),
+    ({"compression": {"method": "lz4"}}, ["-z", "lz4"]),
+    ({"compression": {"method": "lz4", "level": 9}}, ["-z", "lz4,9"]),
+    ({"options": ["dedupe"]}, ["-E", "dedupe"]),
+    ({"options": ["all-fragments", "force-inline-compact"]}, ["-E", "all-fragments,force-inline-compact"]),
```

Comment:
There should be one more test here that uses all options and ensures they are all wired up

Reply:
This is done now, there is an integration test now for this as well.